c/ckb_exec.h

```diff
@@ -0,0 +1,214 @@
+#ifndef _CBK_C_STDLIB_CKB_EXEC_H_
+#define _CBK_C_STDLIB_CKB_EXEC_H_
+#include <stdint.h>
+#include <string.h>
+#include <stdbool.h>
+
+#ifndef CKB_EXEC_MAX_ARGS_COUNT
+#define CKB_EXEC_MAX_ARGS_COUNT 64
+#endif
+
+#ifndef CKB_EXEC_MAX_BUFF_LEN
+#define CKB_EXEC_MAX_BUFF_LEN (32 * 1024)
+#endif
+
+#ifndef CKB_EXEC_MAX_PARAM_LEN
+#define CKB_EXEC_MAX_PARAM_LEN (32 * 1024)
+#endif
+
+enum CkbExecErrorCodeType {
+  ERROR_EXEC_OUT_OF_BOUNDS = 30,
+  ERROR_EXEC_INVALID_HEX,
+};
+
+typedef struct CkbBinaryArgsType {
+  uint32_t count;
+  uint32_t len[CKB_EXEC_MAX_ARGS_COUNT];
+  uint8_t* params[CKB_EXEC_MAX_ARGS_COUNT];
+
+  uint32_t used_buff;
+  uint8_t buff[CKB_EXEC_MAX_BUFF_LEN];
+} CkbBinaryArgsType;
+
+typedef struct CkbHexArgsType {
+  uint32_t used_buff;
+  char buff[CKB_EXEC_MAX_BUFF_LEN];
+} CkbHexArgsType;
+
+static int _exec_getbin(uint8_t x, uint8_t* out) {
+  if (x >= '0' && x <= '9') {
+    *out = x - '0';
+  } else if (x >= 'A' && x <= 'F') {
+    *out = x - 'A' + 10;
+  } else if (x >= 'a' && x <= 'f') {
+    *out = x - 'a' + 10;
+  } else {
+    return ERROR_EXEC_INVALID_HEX;
+  }
+  return 0;
+}
+
+static void _exec_gethex(uint8_t x, char* out) {
+  static char s_mapping[] = {'0', '1', '2', '3', '4', '5', '6', '7',
```

Comment:
https://github.com/nervosnetwork/ckb-production-scripts/blob/auth/c/ckb_exec.h#L51-L56

These similar codes can be combined to reduce the size